### PR TITLE
feat: generate report before $fail built-in executes

### LINKED ISSUE #187: $fail should wait for report to be generated before failing the action
**Is your feature request related to a problem? Please describe.**
Right now, the report is only generated when all the program is executed.
When the `$fail` action is part of the program, the report doesn't get to be generated because upon the execution of the built-in the GitHub action fails altogether and the report doesn't get to be generated.

**Describe the solution you'd like**
Perform the execution with the following order:
1. Generate the report
2. Execute the `$fail` action.

**Describe alternatives you've considered**
*None*

**Additional context**
*None*

### PR DESCRIPTION
**Pull request by shay2025**
## Description

This pull request introduces the change of first generating the report before executing the `$fail` action built-in.
Currently, the behavior is once all actions are executed, then generate the report.
However, when we have the `$fail` action built-in, this action will fail the GitHub action and stop its execution completely and so the report won't be generated if the user has the verbose mode set.

## Related issue

Closes #187 

## Type of change

New feature (non-breaking change which adds functionality)

## How was this tested?

Used the pull request `https://github.com/explore-dev/didactic-robot/pull/47` for testing.
1. Change the reviewpad.yml file in the `https://github.com/explore-dev/didactic-robot` locally
2. On the branch of the proposed changes executed the command: 
  `./main -dry-run=false -github-token=[GITHUB_TOKEN] -pull-request=https://github.com/explore-dev/didactic-robot/pull/47 -reviewpad=../didactic-robot/reviewpad.yml -event-payload=cmd/cli/resources/test_open_pr_event.json`

Besides this, it was checked the unit tests were passing by executing the `task test` cmd.

## Checklist

- [X] I have performed a self-review of my code
- [X] I have ran `task check -f` and have no issues

### CODE REVIEW

File: lang/aladino/interpreter.go [hunk 1]
```diff
@@ -140,6 +141,23 @@ func (i *Interpreter) ExecStatement(statement *engine.Statement) error {
 		return err
 	}
 
+	m := regexp.MustCompile("\\$fail\\(\"(.*)\"\\)")
+
+	isFailAction := m.FindString(statement.Code) != ""
+	if isFailAction {
```
Comment (marcelosousa) [inline lang/aladino/interpreter.go:147]: [reviewpad-sig]: <>
[Link to Comment](https://beta.reviewpad.com/review/github.com/reviewpad/reviewpad/pull/190#comment-ae7a4e31-7e0c-443c-a59b-d9b1ec1c9536)

File: engine/exec.go [hunk 2]
```diff
@@ -37,6 +37,11 @@ func Eval(file *ReviewpadFile, env *Env) (*Program, error) {
 
 	interpreter := env.Interpreter
 
+	var isReviewpadReportEnabled bool
```
Comment (marcelosousa) [inline engine/exec.go:40]: {HIGH} This is just a test comment.

**Review by marcelosousa — COMMENTED** [review]
[reviewpad-sig]: <>
[View on **Reviewpad**](https://beta.reviewpad.com/review/github.com/reviewpad/reviewpad/pull/190?codereviewid=eae62a1d-3c38-4b40-95b5-241b088a3d0e)

**Comment by shay2025** [comment]
Restarting the development of this issue

**Pull request closed without merging**